### PR TITLE
Flax: ignore dtype for configuration

### PR DESCRIPTION
**Pull request by pcuenca**
This makes it possible to save weights and configuration files.

### CODE REVIEW

File: src/diffusers/modeling_flax_utils.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ class FlaxModelMixin:
     """
     config_name = CONFIG_NAME
     _automatically_saved_args = ["_diffusers_version", "_class_name", "_name_or_path"]
-    _flax_internal_args = ["name", "parent"]
+    _flax_internal_args = ["name", "parent", "dtype"]
```
Comment (mishig25) [inline src/diffusers/modeling_flax_utils.py:48]: did you receive a warning while initializing?

**Review by patil-suraj — APPROVED** [review]
LGTM!

File: src/diffusers/modeling_flax_utils.py (hunk 1 above)
Comment (pcuenca) [reply src/diffusers/modeling_flax_utils.py:48]: > did you receive a warning while initializing?

Not about `dtype`. In the case of the VAE, for example, I did the following tests:

- Loaded from the current configuration in `fusing`. Then I got this warning:

```
The config attributes {'architectures': ['encoderKL'], 'double_z': True, 'transformers_version': '4.21.0.dev0'} were passed to FlaxAutoencoderKL, but are not expected and will be ignored. Please verify your config.json configuration file.
```

- Saved the configuration.
- Saved the pretrained weights.
- Loaded from the saved pretrained weights. No warning was emitted here.

**Comment by patrickvonplaten** [comment]
Ok removing for now. Overall I was actually in favor of keeping it in because if someone trains a model with `dtype=bfloat16`, it'd be good to know that. So I think eventually we should store this parameter in the config. However as pointed out by @patil-suraj:

- Should probs be saved as a string
- We need good naming to not confuse the weight dtype with the computation dtype

@pcuenca could you maybe open a "Feature request" issue after merging this that reminds us to eventually save this param?

**Review by patrickvonplaten — APPROVED** [review]

**Pull request merged**